apps/web/server/routers/viewer/slots.tsx

```diff
@@ -59,15 +58,16 @@ const checkForAvailability = ({
     return true;
   }
 
-  const slotEndTime = time.add(eventLength, "minutes");
-  const slotStartTimeWithBeforeBuffer = time.subtract(beforeBufferTime, "minutes");
-  const slotEndTimeWithAfterBuffer = time.add(eventLength + afterBufferTime, "minutes");
+  const slotEndTime = time.add(eventLength, "minutes").utc();
+  const slotStartTimeWithBeforeBuffer = time.subtract(beforeBufferTime, "minutes").utc();
+  const slotEndTimeWithAfterBuffer = time.add(eventLength + afterBufferTime, "minutes").utc();
+
+  return busy.every((busyTime) => {
+    const startTime = dayjs.utc(busyTime.start);
+    const endTime = dayjs.utc(busyTime.end);
```

Comment:
For servers w/ non-UTC timezones, if you call as dayjs() the busyTime.start and busyTime.end are translated to local time. Parsing as UTC parses it as UTC which should work better on those servers.